dev/requirements.txt

```diff
@@ -34,7 +35,9 @@ pydata_sphinx_theme
 ipython
 nbsphinx
 numpydoc
-jinja2<3.0.0
+# Commented this because we need the latest version of jinja2 for Pandas API on Spark.
+# Please manually install jinja2<3.0.0 when building the documentation.
+# jinja2<3.0.0
```

Comment:
> but it's not working for some functionality for Pandas API on Spark from 4.0.0 release.

Which functionality does it not work? I think we should better have one version for development if possible.

Reply:
`(DataFrame|Series).to_latex` does not work without `jinja2>=3.1.2`
```python
>>> import pyspark.pandas as ps
>>> psdf = ps.DataFrame({"A": [1, 2, 3]})
>>> psdf.to_latex()
Traceback (most recent call last):
...
ImportError: Pandas requires version '3.1.2' or newer of 'jinja2' (version '2.11.3' currently installed).
```
But I'm okay we just keep the current status, because this is the only case for now.

Reply:
I think we should probably just leave it as is for now, or let the documentation build works with jinja2 3.0.0+

Reply:
Yeah, I tried to make the build work with jinja2 3.0.0+ at the first time, but there is compatibility issues with Sphinx version we're currently using.

Let me just close this and revisit when the Sphinx support better compatibility with the jinja2 3.0.0+.